### PR TITLE
Updating package to include better typing and extensibility.

### PR DESCRIPTION
**Pull request by TheRealFlyingCoder**
Hey @manosim

I've had a version of Facebook sitting in my remix-auth-socials package for ages and finally found the time to bring it's updates in here.

I'm trying to help so I can re-export each specific social package instead of maintaining the code together. One aspect of that is helping all the other packages align to a similar user experience (typed scopes / fields etc) :)

Let me know if you have any issues with this!

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,74 +1,45 @@
 import {
-  OAuth2Profile,
   OAuth2Strategy,
   OAuth2StrategyVerifyParams,
 } from 'remix-auth-oauth2';
 import type { StrategyVerifyCallback } from 'remix-auth';
 
-import type { FacebookPicture } from './types';
+import type { AdditionalFacebookProfileField, FacebookProfile, FacebookScope, FacebookExtraParams, FacebookStrategyOptions } from './types';
```
Comment (TheRealFlyingCoder) [inline src/index.ts]: Moved all the types into the type file

File: src/index.ts [hunk 2]
```diff
@@ -1,74 +1,45 @@
...
-/**
- * This interface declares what the developer will receive from the strategy
- * to verify the user identity in their system.
- */
-export type FacebookExtraParams = {
-  // FIXME!
-} & Record<string, string | number>;
+export const FacebookName = 'facebook';
+export const FacebookDefaultScopes: FacebookScope[] = ['public_profile', 'email'];
+export const FacebookScopeSeperator = ',';
```
Comment (TheRealFlyingCoder) [inline src/index.ts]: It's better for the user to define scopes as a typed array ['scope1', 'scope2'] and then we do the work to format that into facebook's preferred style

File: src/index.ts [hunk 3]
```diff
@@ -1,74 +1,45 @@
...
+] as const;
 
-/**
- * This interface declares what the developer will receive from the strategy
- * to verify the user identity in their system.
- */
-export type FacebookExtraParams = {
-  // FIXME!
-} & Record<string, string | number>;
+export const FacebookName = 'facebook';
```
Comment (TheRealFlyingCoder) [inline src/index.ts]: This is just a QOL update so my package can re-export your default name

**Comment by TheRealFlyingCoder** [comment]
@manosim just bumping this, hopefully you are available to merge the updates, and release :)

**Comment by TheRealFlyingCoder** [comment]
Hey @manosim  just another friendly bump, hoping you can find the time to merge this and release :) Yours is the only social package remaining!

**Comment by manosim** [comment]
Hi @TheRealFlyingCoder, I will have some availability to take a look on this PR and merge/release it soon. Apologies for the delay!

**Comment by manosim** [comment]
Just had a look and it looks great @TheRealFlyingCoder - will cut a new release soon. Thank you for the PR 🙌

**Pull request merged**